wpengine/faustjs
# Only request token endpoint initially, then use a cookie to determine if there is an authenticated user

## Tasks


## Description

This PR addresses https://github.com/wpengine/faustjs/issues/1723#issuecomment-1896433873 and creates a new cookie that determines if there is a refresh token saved in the cookie, without exposing sensitive data and is accessible in the browser/client.

This allows us to make one initial request to the token endpoint, and then thereafter reference the cookie to determine if there is an authenticated user for stuff like the toolbar. This essentially turns a request for the token on every single page to only once for non-authenticated users.



## Related Issue(s):



## Testing

1. From the `canary` branch, start the example project: `npm run dev -w @faustwp/getting-started-example`
2. Open the network tools, and notice every page you visit has a 401 response for the token endpoint.
3. Now, checkout this branch
4. Run `npm install` at the monorepo root
5. Run `npm run build` at the monorepo root
6. Run the example project `npm run dev -w @faustwp/getting-started-example`
7. With the network tools open, visit the home page http://localhost:3000. Notice the 401 response from the token endpoint. Refresh and see that no additional requests are made to the token endpoint as an unauthenticated user.
8. Try and break it with the other testing things below

### Other testing things to ensure functionality still works as expected

- [ ] Ensure refresh tokens are still set properly on login/logout
- [ ] Ensure backwards compatibility still works. For example, if there is a cookie for just the refresh token, the reference token should still get set properly



## Screenshots



## Documentation Changes



## Dependant PRs
